src/main/resources/Development-documentation/gerrit-sceanrios.md

```diff
@@ -0,0 +1,224 @@
+# Gerrit scenarios
+
+These scenarios describes possible use cases that needs consideration during the
+implementation and update of the plugin
+
+<!-- TOC -->
+
+- [Gerrit scenarios](#gerrit-scenarios)
+  - [Create patch set - CASE1](#create-patch-set---case1)
+  - [Create patch set with rebase - CASE2](#create-patch-set-with-rebase---case2)
+  - [Create patch set with merge - CASE3](#create-patch-set-with-merge---case3)
+  - [Create patch set no previous events - CASE4](#create-patch-set-no-previous-events---case4)
+
+<!-- /TOC -->
+
+## Create patch set - CASE1
+
+Preconditions:
+
+- Previous review submits to  the master with hash `C0`
+- The plugin sends SCS with id `E0`
+
+The user does the following:
+
+- Creates a branch
+- Updates the code
+- Squashes the commits to one commits (`C1`)
+- Pushes to `refs/for/[branch name]`
+- Receives comments from reviewer
+- Updates code
+- Does `commit --amend` (`C2`)
+- Pushes to `refs/for/[branch name]`
+- Gets ok from the reviewer
+- Hits the submit button in Gerrit
+
+Events sent:
+
+- SCC(`E1`) sent for `P1` push with `BASE` link set to `E0`
+- SCC(`E2`) sent for `P2` push with `PREVIOUS_VERSION` link set `E1` and `BASE` link set to `E0`
+- SCS(`E3`) at submit with `CHANGE` link set `E2`
+
+Commit history after submit:
+
+```git
+* C2 Review changes
+* C0 Commit from previous user
+```
+
+Trigger table:
+
+| Event | Triggered by | Comment |
+| ---   | ---          | ---     |
+| E0    |              | Submit  |
+| E1    | P1           |
+| E2    | P2           |
+| E3    |              | Submit  |
+
+SCC Event table:
```

Comment:
Should we have a brief glossary at the top of the file where we spell out SourceChangeCreated and SourceChangeSubmitted? Then we can use the abbreviations :)

Reply:
Added in 041b7cd